influxdb/server.sls

```diff
@@ -77,6 +77,22 @@ influxdb_create_db_{{db.name}}:
       - cmd: influxdb_create_admin
     {%- endif %}
   # TODO: manage database deletion
+
+    {%- if db.get('retention_policy') is mapping %}
+    {% set rp_name = db.retention_policy.get('name', 'autogen') %}
+    {% set duration = db.retention_policy.get('duration', 'INF') %}
+    {% set replication = db.retention_policy.get('replication', '1') %}
+    {% set shard_duration = db.retention_policy.get('shard_duration', False) %}
+    {% set query_rentention_policy = 'RETENTION POLICY {} ON {} DURATION {} REPLICATION {}'.format(
+        rp_name, db.name, duration, replication)
+    %}
+influxdb_retention_policy_{{db.name}}:
+  cmd.run:
+
+    - name: curl -s -S -POST "{{ url_for_query }}" --data-urlencode "q=CREATE {{ query_rentention_policy }}{%- if shard_duration %} SHARD DURATION {{ shard_duration }}{%- endif %}"|grep "policy already exists" || curl -s -S -POST "{{ url_for_query }}" --data-urlencode "q=ALTER {{ query_rentention_policy }}{%- if shard_duration %} SHARD DURATION {{ shard_duration }}{%- endif %}"
```

Comment:
not sure why "SHARD DURATION ..." isn't added to `query_rentention_policy` before.